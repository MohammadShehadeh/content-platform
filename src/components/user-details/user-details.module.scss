@use '@/styles/shared/breakpoints' as breakpoints;

.header {
  margin-bottom: var(--space-md);
  display: flex;
  gap: var(--space-sm);
  flex-direction: column;
  justify-content: flex-start;
  align-items: start;

  @include breakpoints.respond-above('sm') {
    flex-direction: row;
    gap: var(--space-md);
    align-items: center;
  }
}

.avatar {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 64px;
  height: 64px;
  border-radius: 50%;
  background-color: hsl(var(--primary) / 0.1);
  color: hsl(var(--primary));
}

.section {
  margin-bottom: var(--space-lg);

  &:last-child {
    margin-bottom: 0;
  }
}

.infoGrid {
  columns: 2;
}

.infoItem {
  display: flex;
  align-items: flex-start;
  gap: var(--space-sm);
  margin-bottom: var(--space-md);

  &:last-child {
    margin-bottom: 0;
  }
}

.icon {
  width: 20px;
  height: 20px;
  color: hsl(var(--primary));
  margin-top: 2px;
  flex-shrink: 0;
}

.label {
  display: block;
  font-size: var(--font-size-sm);
  color: hsl(var(--muted-foreground));
  margin-bottom: 4px;
}

.value {
  font-size: var(--font-size-md);
  color: hsl(var(--foreground));
  text-decoration: none;

  &:hover {
    color: hsl(var(--primary));
    text-decoration: underline;
  }
}

.addressLine {
  margin: 0;
  margin-bottom: var(--space-xs);
  font-size: var(--font-size-md);

  &:last-child {
    margin-bottom: 0;
  }
}
