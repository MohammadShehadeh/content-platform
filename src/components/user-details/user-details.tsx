import { Mail, Phone, Globe, MapPin, Briefcase } from 'lucide-react';

import { Avatar } from '@/components/avatar';
import { Description } from '@/components/description';
import { Title } from '@/components/title';
import type { User } from '@/services/users';

import styles from './user-details.module.scss';

interface UserDetailsProps {
  user: User;
}

export const UserDetails = ({ user }: UserDetailsProps) => {
  return (
    <div className={styles.container}>
      <header className={styles.header}>
        <Avatar />
        <div>
          <Title>{user.name}</Title>
          <Description color="muted">@{user.username}</Description>
        </div>
      </header>

      <div className={styles.infoGrid}>
        <div className={styles.infoItem}>
          <Mail className={styles.icon} />
          <a href={`mailto:${user.email}`} className={styles.value}>
            {user.email}
          </a>
        </div>
        <div className={styles.infoItem}>
          <Phone className={styles.icon} />
          <a href={`tel:${user.phone}`} className={styles.value}>
            {user.phone}
          </a>
        </div>

        <div className={styles.infoItem}>
          <MapPin className={styles.icon} />
          <Description size="lg" disableGutter>
            {user.address.street}, {user.address.suite} {user.address.city},{' '}
            {user.address.zipcode}
          </Description>
        </div>
        <div className={styles.infoItem}>
          <Globe className={styles.icon} />
          <a
            href={`https://${user.website}`}
            target="_blank"
            rel="noopener noreferrer"
            className={styles.value}
          >
            {user.website}
          </a>
        </div>
        <div className={styles.infoItem}>
          <Briefcase className={styles.icon} />
          <div>
            <Description size="lg" disableGutter>
              {user.company.name}
            </Description>
            <Description color="muted" disableGutter>
              {user.company.catchPhrase}
            </Description>
            <Description color="muted" disableGutter>
              {user.company.bs}
            </Description>
          </div>
        </div>
      </div>
    </div>
  );
};
