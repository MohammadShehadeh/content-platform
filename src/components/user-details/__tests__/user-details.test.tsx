import { render, screen } from '@testing-library/react';

import { UserDetails } from '@/components/user-details';

describe('UserDetails', () => {
  const mockUser = {
    id: 1,
    name: 'John Doe',
    username: 'johndoe',
    email: 'john@example.com',
    phone: '555-1234',
    website: 'example.com',
    address: {
      street: '123 Main St',
      suite: 'Apt 4',
      city: 'New York',
      zipcode: '10001',
      geo: { lat: '40.7128', lng: '-74.0060' },
    },
    company: {
      name: 'Example Corp',
      catchPhrase: 'Making examples since 2023',
      bs: 'innovative solutions',
    },
  };

  it('should render user information correctly', () => {
    render(<UserDetails user={mockUser} />);

    expect(screen.getByText('John Doe')).toBeInTheDocument();
    expect(screen.getByText('@johndoe')).toBeInTheDocument();
    expect(screen.getByText('john@example.com')).toBeInTheDocument();
    expect(screen.getByText('555-1234')).toBeInTheDocument();
    expect(screen.getByText('example.com')).toBeInTheDocument();
    expect(screen.getByText('Example Corp')).toBeInTheDocument();
    expect(screen.getByText('123 Main St, Apt 4 New York, 10001')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<UserDetails user={mockUser} />);
    expect(container).toMatchSnapshot();
  });
});
