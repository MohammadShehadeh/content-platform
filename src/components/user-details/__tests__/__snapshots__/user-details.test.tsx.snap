// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails matches snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <header
      class="header"
    >
      <div
        class="avatar"
        data-testid="avatar"
      >
        <svg
          class="lucide lucide-user"
          fill="none"
          height="32"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 21v-2a4 4 0 0 0-4-4H9a4 4 0 0 0-4 4v2"
          />
          <circle
            cx="12"
            cy="7"
            r="4"
          />
        </svg>
      </div>
      <div>
        <h1
          class="title lg"
        >
          John Doe
        </h1>
        <p
          class="description md muted"
        >
          @
          johndoe
        </p>
      </div>
    </header>
    <div
      class="infoGrid"
    >
      <div
        class="infoItem"
      >
        <svg
          class="lucide lucide-mail icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="16"
            rx="2"
            width="20"
            x="2"
            y="4"
          />
          <path
            d="m22 7-8.97 5.7a1.94 1.94 0 0 1-2.06 0L2 7"
          />
        </svg>
        <a
          class="value"
          href="mailto:john@example.com"
        >
          john@example.com
        </a>
      </div>
      <div
        class="infoItem"
      >
        <svg
          class="lucide lucide-phone icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M22 16.92v3a2 2 0 0 1-2.18 2 19.79 19.79 0 0 1-8.63-3.07 19.5 19.5 0 0 1-6-6 19.79 19.79 0 0 1-3.07-8.67A2 2 0 0 1 4.11 2h3a2 2 0 0 1 2 1.72 12.84 12.84 0 0 0 .7 2.81 2 2 0 0 1-.45 2.11L8.09 9.91a16 16 0 0 0 6 6l1.27-1.27a2 2 0 0 1 2.11-.45 12.84 12.84 0 0 0 2.81.7A2 2 0 0 1 22 16.92z"
          />
        </svg>
        <a
          class="value"
          href="tel:555-1234"
        >
          555-1234
        </a>
      </div>
      <div
        class="infoItem"
      >
        <svg
          class="lucide lucide-map-pin icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 10c0 4.993-5.539 10.193-7.399 11.799a1 1 0 0 1-1.202 0C9.539 20.193 4 14.993 4 10a8 8 0 0 1 16 0"
          />
          <circle
            cx="12"
            cy="10"
            r="3"
          />
        </svg>
        <p
          class="description lg default disableGutter"
        >
          123 Main St
          , 
          Apt 4
           
          New York
          ,
           
          10001
        </p>
      </div>
      <div
        class="infoItem"
      >
        <svg
          class="lucide lucide-globe icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <path
            d="M12 2a14.5 14.5 0 0 0 0 20 14.5 14.5 0 0 0 0-20"
          />
          <path
            d="M2 12h20"
          />
        </svg>
        <a
          class="value"
          href="https://example.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          example.com
        </a>
      </div>
      <div
        class="infoItem"
      >
        <svg
          class="lucide lucide-briefcase icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 20V4a2 2 0 0 0-2-2h-4a2 2 0 0 0-2 2v16"
          />
          <rect
            height="14"
            rx="2"
            width="20"
            x="2"
            y="6"
          />
        </svg>
        <div>
          <p
            class="description lg default disableGutter"
          >
            Example Corp
          </p>
          <p
            class="description md muted disableGutter"
          >
            Making examples since 2023
          </p>
          <p
            class="description md muted disableGutter"
          >
            innovative solutions
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`UserDetails should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <header
      class="header"
    >
      <div
        class="avatar"
        data-testid="avatar"
      >
        <svg
          class="lucide lucide-user"
          fill="none"
          height="32"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 21v-2a4 4 0 0 0-4-4H9a4 4 0 0 0-4 4v2"
          />
          <circle
            cx="12"
            cy="7"
            r="4"
          />
        </svg>
      </div>
      <div>
        <h1
          class="title lg"
        >
          John Doe
        </h1>
        <p
          class="description md muted"
        >
          @
          johndoe
        </p>
      </div>
    </header>
    <div
      class="infoGrid"
    >
      <div
        class="infoItem"
      >
        <svg
          class="lucide lucide-mail icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="16"
            rx="2"
            width="20"
            x="2"
            y="4"
          />
          <path
            d="m22 7-8.97 5.7a1.94 1.94 0 0 1-2.06 0L2 7"
          />
        </svg>
        <a
          class="value"
          href="mailto:john@example.com"
        >
          john@example.com
        </a>
      </div>
      <div
        class="infoItem"
      >
        <svg
          class="lucide lucide-phone icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M22 16.92v3a2 2 0 0 1-2.18 2 19.79 19.79 0 0 1-8.63-3.07 19.5 19.5 0 0 1-6-6 19.79 19.79 0 0 1-3.07-8.67A2 2 0 0 1 4.11 2h3a2 2 0 0 1 2 1.72 12.84 12.84 0 0 0 .7 2.81 2 2 0 0 1-.45 2.11L8.09 9.91a16 16 0 0 0 6 6l1.27-1.27a2 2 0 0 1 2.11-.45 12.84 12.84 0 0 0 2.81.7A2 2 0 0 1 22 16.92z"
          />
        </svg>
        <a
          class="value"
          href="tel:555-1234"
        >
          555-1234
        </a>
      </div>
      <div
        class="infoItem"
      >
        <svg
          class="lucide lucide-map-pin icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 10c0 4.993-5.539 10.193-7.399 11.799a1 1 0 0 1-1.202 0C9.539 20.193 4 14.993 4 10a8 8 0 0 1 16 0"
          />
          <circle
            cx="12"
            cy="10"
            r="3"
          />
        </svg>
        <p
          class="description lg default disableGutter"
        >
          123 Main St
          , 
          Apt 4
           
          New York
          ,
           
          10001
        </p>
      </div>
      <div
        class="infoItem"
      >
        <svg
          class="lucide lucide-globe icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <path
            d="M12 2a14.5 14.5 0 0 0 0 20 14.5 14.5 0 0 0 0-20"
          />
          <path
            d="M2 12h20"
          />
        </svg>
        <a
          class="value"
          href="https://example.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          example.com
        </a>
      </div>
      <div
        class="infoItem"
      >
        <svg
          class="lucide lucide-briefcase icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16 20V4a2 2 0 0 0-2-2h-4a2 2 0 0 0-2 2v16"
          />
          <rect
            height="14"
            rx="2"
            width="20"
            x="2"
            y="6"
          />
        </svg>
        <div>
          <p
            class="description lg default disableGutter"
          >
            Example Corp
          </p>
          <p
            class="description md muted disableGutter"
          >
            Making examples since 2023
          </p>
          <p
            class="description md muted disableGutter"
          >
            innovative solutions
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
